src/base/bittorrent/torrentinfo.cpp

```diff
@@ -167,18 +167,25 @@ TorrentInfo TorrentInfo::loadFromFile(const QString &path, QString *error) noexc
 void TorrentInfo::saveToFile(const QString &path) const
 {
     if (!isValid())
-        throw RuntimeError {tr("Invalid metadata.")};
+        throw RuntimeError {tr("Invalid metadata")};
 
-    const lt::create_torrent torrentCreator = lt::create_torrent(*(nativeInfo()));
-    const lt::entry torrentEntry = torrentCreator.generate();
+    try
+    {
+        const lt::create_torrent torrentCreator {*(nativeInfo())};
```

Comment:
IIRC you disliked mixing different initializing styles (Uniform initialization and assignment) and now you are fine with it?

Reply:
When we're talking about identical operations.
Here's a slightly different case, IMO. One line creates the object directly. The other makes a copy of the object returned from the function.
What I really dislike is explicitly using the copy constructor:
```c++
const lt::entry torrentEntry {torrentCreator.generate()};
```
But if you want, I could rewrite it like this:
```c++
const auto torrentCreator = lt::create_torrent(*(nativeInfo()));
const lt::entry torrentEntry = torrentCreator.generate();
```

Reply:
>Here's a slightly different case, IMO. One line creates the object directly. The other makes a copy of the object returned from the function.

I don't differentiate that much, all I see is a constructor interface for an object. Whether it is copy or moved-in or anything else will depend on the implementation of the chosen constructor.

>But if you want, I could rewrite it like this:

I'm not asking for some changes here, just wondering your viewpoint on it. And sure it is more readable in my eyes if they used similar construct syntax.

Reply:
> I don't differentiate that much, all I see is a constructor interface for an object. Whether it is copy or moved-in or anything else will depend on the implementation of the chosen constructor.

Okay, forget it. Mixing initialization styles next to each other is really not a good idea.
Changed.